### PR TITLE
Googletest Update

### PR DESCRIPTION
**Pull request by chapman39**

### CODE REVIEW

File: thirdparty_builtin/CMakeLists.txt [hunk 1]
```diff
@@ -7,9 +7,9 @@ set(_blt_tpl_targets) # tracks names of enabled tpl targets
 
 if(ENABLE_GBENCHMARK OR ENABLE_GTEST OR ENABLE_GMOCK)
   if("${CMAKE_CXX_STANDARD}" STREQUAL "" OR
-     "${CMAKE_CXX_STANDARD}" LESS 11)
+     "${CMAKE_CXX_STANDARD}" LESS 14)
       message(WARNING "GoogleTest, GoogleMock, and GoogleBenchmark require C++11 or later. "
```
Comment (white238) [inline thirdparty_builtin/CMakeLists.txt:11]:
```suggestion
      message(WARNING "GoogleTest, GoogleMock, and GoogleBenchmark require C++14 or later. "
```

**Comment by white238** [comment]
Please update `RELEASE-NOTES.md` and make sure to note what date this was taken from googletest and what branch.

**Comment by chapman39** [comment]
@white238 do you want me to keep this?
https://github.com/LLNL/blt/blob/fa4dda8154d134c9464a4f6a31d1ccfbcebc678c/thirdparty_builtin/googletest_version.txt

**Review by white238 — APPROVED** [review]

**Comment by chapman39** [comment]
`Error encountered during job: mkdir /g/g16/atk/.jacamar-ci/scripts/zkQzXxmM/002/blt/blt/1142394: disk quota exceeded`

**Comment by adrienbernede** [comment]
@chapman39 since you restricted your patch to xl, my error is back on intel...

File: host-configs/llnl/toss_3_x86_64_ib/intel-oneapi@2021.3.cmake [hunk 2]
```diff
@@ -31,6 +31,21 @@ set(ENABLE_FORTRAN ON CACHE BOOL "")
...
+set(BLT_CXX_STD "c++14")
+
+# Use non-system default stdlib
+set(GCC_VERSION "gcc-9.3.1")
+# Note: gcc-8.3.1 and above need the "/rh" directory. Below that version do not.
+set(GCC_HOME "/usr/tce/packages/gcc/${GCC_VERSION}/rh")
+
+set(BLT_C_FLAGS "--gcc-toolchain=${GCC_HOME}" CACHE STRING "")
+set(BLT_CXX_FLAGS "--gcc-toolchain=${GCC_HOME}" CACHE STRING "")
+set(BLT_FORTRAN_FLAGS "-gcc-name=${GCC_HOME}/bin/gcc" CACHE STRING "")
```
Comment (adrienbernede) [inline host-configs/llnl/toss_3_x86_64_ib/intel-oneapi@2021.3.cmake:47]: Interesting this change of flag for the fortran compiler. I’m gonna try that to see if it solves some issue I have. What’s the process that got you there? I mean, did you copy some existing conf, or did you check the compiler options?

**Review by adrienbernede — APPROVED** [review]

File: host-configs/llnl/toss_3_x86_64_ib/intel-oneapi@2021.3.cmake (hunk 2 above)
Comment (adrienbernede) [reply host-configs/llnl/toss_3_x86_64_ib/intel-oneapi@2021.3.cmake:47]: I see where this comes from: intel is now relying on the llvm compiler, except for frotran. So the options remain the same for ifx, and that should be -gcc-name.
Unfortunately, that did not change anything to the error I have with ifx ;).

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -9,6 +9,9 @@ The project release numbers follow [Semantic Versioning](http://semver.org/spec/
 
 ## [Unreleased] - Release date yyyy-mm-dd
 
+### Changed
+- Updated Googletest to main from 04/13/2023. Commit: [12a5852e451baabc79c63a86c634912c563d57bc](https://github.com/google/googletest/commit/12a5852e451baabc79c63a86c634912c563d57bc)
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:13]: Please note the C++ version required for the new googletest, and that PGI is no longer supported (presumably, when configured with `ENABLE_GTEST`).

If any users require PGI, we might need to patch gtest to make this work.

File: thirdparty_builtin/googletest_version.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Googletest version is from 04/13/2023
+https://github.com/google/googletest/commit/12a5852e451baabc79c63a86c634912c563d57bc
```
Comment (kennyweiss) [inline thirdparty_builtin/googletest_version.txt:2]: 👍

**Review by kennyweiss — APPROVED** [review]
Thanks @chapman39 and @white238 and @adrienbernede !

File: RELEASE-NOTES.md (hunk 3 above)
Comment (white238) [reply RELEASE-NOTES.md:13]: I asked the only internal user I knew about that used PGI and they had left the legacy PGI compiler and hadn't even tried the newer one ("nvhpc") yet. 

I say we go with not testing it until a later date.

Comment (kennyweiss) [reply RELEASE-NOTES.md:13]: I agree, but we should note this in our `RELEASE-NOTES`.

Comment (chapman39) [reply RELEASE-NOTES.md:13]: I added:

> Note: this version of Googletest requires C++14, and PGI is not supported. If you are using PGI, set ENABLE_GTEST OFF.

Comment (chapman39) [reply RELEASE-NOTES.md:13]: I forgot to refresh the page. I can make note of this in the RELEASE-NOTES as well

Comment (kennyweiss) [reply RELEASE-NOTES.md:13]: Thanks!

**Pull request merged**